internal/testing/cmdtest/cmdtest.go

```diff
@@ -329,6 +297,18 @@ func (ts *TestSuite) compare(files []*testFile) error {
 	return nil
 }
 
+// CompareT is similar to Compare, but runs each test file in a separate
+// subtest, and calls t.Error if there are errors or differences.
+func (ts *TestSuite) CompareT(t *testing.T) {
+	for _, tf := range ts.files {
+		t.Run(tf.filename, func(t *testing.T) {
+			if s := tf.compare(); s != "" {
+				t.Error(s)
+			}
+		})
+	}
+}
+
 func (tf *testFile) compare() string {
 	if err := tf.execute(); err != nil {
 		return fmt.Sprintf("%v\n", err)
```

Comment:
nit: I think you can remove the terminating newline here and below, since the output is passed to `t.Error`.

Reply:
Done.